### PR TITLE
Cleanup user session on app start

### LINKED ISSUE #351: unexpected behavior when logging out
reproduce: 
1) install app
2) Continue with Gmail, and authenticate with account A
3) while it's loading backups on setup screen, kill the app (maybe the same could be achieved by waiting until it's done loading, then choosing "use another account". The point is that the app will receive and save access tokens for account A, but setup won't be finished)
4) re-open the app to again see "Continue with Gmail". Now continue with account B
5) actually finish setup of account B
6) menu -> log out

expected behavior: I should see splash screen with "Continue with Gmail"

actual behavior:  takes me to "Set up FlowCrypt" where it loads backups for account A (I'm not expecting this after I just logged out of account B)

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR Fix issue in case user hasn't finish setup and kill the app

close #351 

----------------------------------

**Tests**:
- Difficult to test. This part requires to have all other parts to be covered with tests
- Manual test

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Review by tomholub — COMMENTED** [review]
Code looks good, will re-test

**Comment by Kharchevskyi** [comment]
is it possible to prioritise #323  before merging this PR?

**Comment by tomholub** [comment]
Yes - definitely

**Comment by tomholub** [comment]
Please ping me once it has been brought up to date / conflicts resolved. Thanks!

**Comment by Kharchevskyi** [comment]
Merged master. I haven't test all flows if it works with "new setup flow", but I assume it will. 
(_Going to check it probably tomorrow, or you can check and merge if everything works as expected_)

File: FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift [hunk 1]
```diff
@@ -183,10 +183,7 @@ extension EncryptedStorage {
     }
 
     func isAnyKey(for email: String) -> Bool {
```
Comment (tomholub) [inline FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift:185]: Should rename `isAnyKey` to  `doesAnyKeyExist` - filing another issue

**Review by tomholub — APPROVED** [review]
code ok still, will check functionality now

**Comment by tomholub** [comment]
works well

**Pull request merged**